### PR TITLE
Version 5 Upgrade docs updated for flex_template_job, node_group, and import format regex

### PR DESCRIPTION
**Pull request by NickElliot**
flex_template: https://github.com/GoogleCloudPlatform/magic-modules/pull/9031
node_group: https://github.com/GoogleCloudPlatform/magic-modules/pull/9044
import regex: https://github.com/GoogleCloudPlatform/magic-modules/pull/8913

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by NickElliot** [comment]
[output.log](https://github.com/GoogleCloudPlatform/magic-modules/files/12704310/output.log)
In the interest of not delaying for CI run for review, here is the updated test passing locally

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown [hunk 1]
```diff
@@ -586,6 +589,31 @@ Use the `google_identity_platform_config` resource instead. It contains a more c
 
 `reconcile_connections` previously defaults to true. Now it will default from the API.
 
+
+## Resource: `google_dataflow_flex_template_job`
```
Comment (c2thorn) [inline mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown:593]: adding @rileykarson to LGTM this section

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown [hunk 2]
```diff
@@ -586,6 +589,31 @@ Use the `google_identity_platform_config` resource instead. It contains a more c
...
+
+### Fields that are a part of the [environment block](https://cloud.google.com/dataflow/docs/reference/rest/v1b3/projects.locations.flexTemplates/launch#FlexTemplateRuntimeEnvironment) will be overriden to be sent via their fields even when supplied via parameters.
+
+Several fields within the `flex_template_job` resource can be supplied as part of the `parameters{}` block rather than configured directly as updates to their corresponding fields, which had Terraform support added in version 4.66. To resolve a conflict resulting from both values being included within Terraform state, all supplied key-value pair parameters with corresponding fields will be overridden to be sent to their direct field endpoint rather than included as part of `parameters{}` in the post request.
+
+Additionally, due to the API returning these fields to the user they will now be considered computed and users will see values twice within their state if they wish to continue supplying these fields' values via parameters.
```
Comment (rileykarson) [inline mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown:599]:
```suggestion
Additionally, due to the API returning these fields to the user they will now be considered computed and users will see values twice within their state when configuring these fields' values via the `parameters{}` block.
```

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown [hunk 3]
```diff
@@ -586,6 +589,31 @@ Use the `google_identity_platform_config` resource instead. It contains a more c
 
 `reconcile_connections` previously defaults to true. Now it will default from the API.
 
+
+## Resource: `google_dataflow_flex_template_job`
+
+### Fields that are a part of the [environment block](https://cloud.google.com/dataflow/docs/reference/rest/v1b3/projects.locations.flexTemplates/launch#FlexTemplateRuntimeEnvironment) will be overriden to be sent via their fields even when supplied via parameters.
+
+Several fields within the `flex_template_job` resource can be supplied as part of the `parameters{}` block rather than configured directly as updates to their corresponding fields, which had Terraform support added in version 4.66. To resolve a conflict resulting from both values being included within Terraform state, all supplied key-value pair parameters with corresponding fields will be overridden to be sent to their direct field endpoint rather than included as part of `parameters{}` in the post request.
```
Comment (rileykarson) [inline mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown:597]:
```suggestion
Several fields within the `google_dataflow_flex_template_job` resource can be supplied through either the `parameters{}` block or a field on the resource object. Support for these fields on the resource object was added in the `4.66.0` release of the Google provider. That version introduced an issue where the values were being double-sent to the API due to being recorded in Terraform state in two places. To resolve this issue, these fields will be deduplicated and sent to the API through the resource object.
```

**Review by rileykarson — COMMENTED** [review]
Rewording suggestions inline

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown (hunk 1 above)
Comment (rileykarson) [reply mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown:593]: LGTM (gonna leave clicking the green check to you)

**Review by c2thorn — APPROVED** [review]

**Pull request merged**